app/react/portainer/HomeView/EnvironmentList/EnvironmentItem/AgentDetails.tsx

```diff
@@ -1,10 +1,39 @@
+import { AlertTriangle } from 'lucide-react';
+
 import { Environment } from '@/react/portainer/environments/types';
-import { isAgentEnvironment } from '@/react/portainer/environments/utils';
+import {
+  isAgentEnvironment,
+  isEdgeEnvironment,
+} from '@/react/portainer/environments/utils';
+import { isVersionSmaller } from '@/react/common/semver-utils';
+import { useSystemStatus } from '@/react/portainer/system/useSystemStatus';
+
+import { Tooltip } from '@@/Tip/Tooltip';
+import { Icon } from '@@/Icon';
 
 export function AgentDetails({ environment }: { environment: Environment }) {
-  if (!isAgentEnvironment(environment.Type)) {
+  const { data: systemStatus } = useSystemStatus();
+
+  if (!systemStatus || !isAgentEnvironment(environment.Type)) {
     return null;
   }
 
-  return <span>{environment.Agent.Version}</span>;
+  const { Version } = systemStatus;
+  const isSmallerEdge =
+    environment.Agent.Version &&
```

Comment:
if `Version` is empty we can have two cases:
version didn't arrive yet or the agent version is so small (I think <2.16) that the agent doesn't send it.

for the first option, I think we can check if the endpoint has any snapshots. for the second option we should show the notification

Reply:
Snapshots are only docker snapshots, they don't contain the Agent version. Also if the env has a snapshot, it will have the version available.

Reply:
unless the agent version is too old